server.go

```diff
@@ -87,6 +89,11 @@ func (srv *Server) Handler(channel string) http.HandlerFunc {
 			return
 		}
 
+		var maxConnTimeCh <-chan time.Time
+		if srv.MaxConnTime > 0 {
+			maxConnTimeCh = time.After(srv.MaxConnTime)
```

Comment:
[`time.After`](https://golang.org/pkg/time/#After) will prevent the timer which is used to send to the channel from being garbage collected until it expires. Since this is called for every request, I think you'll want to manually clean-up the timer so that they do not outlive the request lifetime. Something like the following should work.

```suggestion
			t := time.NewTimer(srv.MaxConnTime)
			defer t.Stop()
			maxConnTimeCh = t.C
```

Reply:
I always forget about that, thanks.